src/Workspaces/CSharp/Portable/Extensions/CastExpressionSyntaxExtensions.cs

```diff
@@ -162,43 +162,40 @@ private static bool UserDefinedConversionIsAllowed(ExpressionSyntax expression,
             //
             // Foo((object)null);
 
-            if (cast.Expression.WalkDownParentheses().IsKind(SyntaxKind.NullLiteralExpression))
+            var argument = cast.WalkUpParentheses().Parent as ArgumentSyntax;
+            if (argument != null)
             {
-                var argument = cast.WalkUpParentheses().Parent as ArgumentSyntax;
-                if (argument != null)
+                // If there are any arguments to the right, we can assume that this is not a
+                // *single* argument passed to a params parameter.
+                var argumentList = argument.Parent as BaseArgumentListSyntax;
+                if (argumentList != null)
                 {
-                    // If there are any arguments to the right, we can assume that this is not a
-                    // *single* argument passed to a params parameter.
-                    var argumentList = argument.Parent as BaseArgumentListSyntax;
-                    if (argumentList != null)
+                    var argumentIndex = argumentList.Arguments.IndexOf(argument);
+                    if (argumentIndex < argumentList.Arguments.Count - 1)
                     {
-                        var argumentIndex = argumentList.Arguments.IndexOf(argument);
-                        if (argumentIndex < argumentList.Arguments.Count - 1)
-                        {
-                            return false;
-                        }
+                        return false;
                     }
+                }
 
-                    var parameter = argument.DetermineParameter(semanticModel, cancellationToken: cancellationToken);
-                    if (parameter != null && parameter.IsParams)
-                    {
-                        Debug.Assert(parameter.Type is IArrayTypeSymbol);
+                var parameter = argument.DetermineParameter(semanticModel, cancellationToken: cancellationToken);
+                if (parameter != null && parameter.IsParams)
+                {
+                    Debug.Assert(parameter.Type is IArrayTypeSymbol);
```

Comment:
I have no idea. I just left the rest of the code the same. I'll take a closer look. Thanks!